src/middleware/authMiddleware.ts

```diff
@@ -5,33 +5,114 @@ import { JwtPayload } from "../types/auth";
 import { User } from "../models/User";
 import { AppDataSource } from "../config/db";
 
-declare global {
-  namespace Express {
-    interface Request {
-      user?: JwtPayload;
-    }
-  }
-}
-
-export const authenticateToken = (
+export const authenticateToken = async (
   req: Request,
   res: Response,
   next: NextFunction
 ) => {
-  const authHeader = req.headers.authorization;
-  const token = authHeader && authHeader.split(" ")[1];
+
+  const authHeader = 
+    req.headers['authorization'] || 
+    req.headers['Authorization'] || 
+    req.headers['AUTHORIZATION'];
+
+  console.log(`Raw Authorization Header: ${authHeader}`);
+
+  if (!authHeader) {
+    console.error('No authorization header found');
+    return res.status(401).json({ 
+      message: "Unauthorized: No authorization header",
+      details: "Authorization header is missing" 
+    });
+  }
+
+  const normalizedHeader = authHeader.toString().trim();
+  
+  console.log(`Normalized Authorization Header: ${normalizedHeader}`);
+
+  const tokenMatch = normalizedHeader.match(/^Bearer\s+(.+)$/i);
+  
+  if (!tokenMatch) {
+    console.error('Invalid authorization header format');
+    return res.status(401).json({ 
+      message: "Unauthorized: Invalid authorization header format",
+      details: "Header must start with 'Bearer ' followed by token" 
+    });
+  }
+
+  const token = tokenMatch[1].trim();
 
   if (!token) {
-    return res.status(401).json({ message: "Unauthorized" });
+    console.error('Token is empty');
+    return res.status(401).json({ 
+      message: "Unauthorized: Token is empty",
+      details: "No token provided after 'Bearer '" 
+    });
   }
 
   try {
     const decoded = verifyToken(token, env.JWT_ACCESS_SECRET) as JwtPayload;
-    req.user = decoded;
+
+    if (!decoded || !decoded.id) {
+      console.error('Invalid token payload', decoded);
+      return res.status(401).json({ 
+        message: "Unauthorized: Invalid token payload",
+        details: "Token is missing required information" 
+      });
+    }
+
+    const userRepo = AppDataSource.getRepository(User);
+    const user = await userRepo.findOne({
+      where: { id: decoded.id },
+      relations: ["roles"],
+    });
+
+    if (!user) {
+      console.error(`User not found for ID: ${decoded.id}`);
+      return res.status(401).json({ 
+        message: "Unauthorized: User not found",
+        details: `No user exists with ID ${decoded.id}` 
```

Comment:
The detail is not needed. the message alone is enough

Reply:
Okay, sure...